mltrace/entities/base_component.py

```diff
@@ -269,13 +270,34 @@ def wrapper(*args, **kwargs):
                         should_filter=True, **all_input_args
                     )
 
+                def mlflow_start_run_id():
+                    nonlocal mlflow_run_id
+                    res = mlflow_start_run_copy()
+                    if mlflow.active_run():
+                        mlflow_run_id = mlflow.active_run().info.run_id
+                    return res
+
+                mlflow_run_id = None
+                mlflow_start_run_copy = mlflow.start_run
+                mlflow.start_run = mlflow_start_run_id
+                # monkey patching mlflow.start_run method
+
                 component_run.set_start_timestamp()
                 # Run function
                 local_vars, value = utils.run_func_capture_locals(
                     func, *args, **kwargs
                 )
                 component_run.set_end_timestamp()
 
+                if mlflow_run_id is not None:
```

Comment:
Do some error checking here / do this in a try-catch & emit a warning if `mlflow.get_run` fails